dotcom-rendering/src/web/server/extractIslands.ts

```diff
@@ -0,0 +1,21 @@
+const regex = new RegExp(
+	// Gets the name of any island where expediteLoading is set to 'true'
+	// e.g <gu-island name="MyComponent" props="..." expediteLoading="true"></gu-island>
+	/<gu-island name="([a-zA-Z]+)"[^>]*expediteLoading="true"[^>]*><\/gu-island>/g,
```

Comment:
Do we want to [start introducing `RegExp` for html](https://blog.codinghorror.com/parsing-html-the-cthulhu-way/)? I’m tempted to lean towards a proper parser like `sanitize-html`, which we currently use, or `jsdom`, and simply call a `querySelectorAll(gu-island[defer="critical"])` on it.

Reply:
That’s a valid point… but then again we’re not really measuring any of this, so it’s hard to make an educated guess.

Running the code below on a recent article, it seems that parsing the DOM takes about 6ms on average. I could easily parse it 5000+ times before getting a crash

```js
const html = await fetch(
  "https://www.theguardian.com/football/2022/may/16/chelsea-have-robert-lewandowski-on-radar-despite-favoured-barcelona-move"
).then((r) => r.text());

const parser = new DOMParser();
const doms = [];
performance.mark("parsing-start");
for (let index = 0; index < 1_000; index++) {
  const dom = parser.parseFromString(html, "text/html");
  doms.push(dom);
}
performance.measure("parsing", "parsing-start");
```

Reply:
Awesome! I shall switch over to this then :)